src/utils/isModelValueType.ts

```diff
@@ -1,10 +1,12 @@
 import { ModelValueType, PrimitiveValueType } from '../glossary'
+import { isObject } from './isObject'
 
 function isPrimitiveValueType(value: any): value is PrimitiveValueType {
   return (
     typeof value === 'string' ||
     typeof value === 'number' ||
     typeof value === 'boolean' ||
+    isObject(value) ||
```

Comment:
How will this work with nested objects? Those will be treated as primitive value types with this change, preventing the library from recursively analyzing their fields.

Reply:
Our test suite will confirm that. Will run it to verify...